cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/PipelineRegisterer.java

```diff
@@ -106,22 +114,41 @@ public Pipeline registerPlugins(ETLConfig config, Class errorDatasetType, Datase
       }
       PipelineConfigurer transformConfigurer = new DefaultPipelineConfigurer(configurer, transformId);
       transformObj.configurePipeline(transformConfigurer);
-      transformIds.add(new TransformInfo(transformId, transformConfig.getErrorDatasetName()));
+      transformInfos.add(new TransformInfo(transformId, transformConfig.getErrorDatasetName()));
       transforms.add(transformObj);
+
+      pluginNum++;
     }
 
-    // Validate Source -> Transform -> Sink hookup
-    try {
-      validateStages(source, sink, transforms);
-      PipelineConfigurer sourceConfigurer = new DefaultPipelineConfigurer(configurer, sourcePluginId);
+    List<String> sinkPluginIds = new ArrayList<>();
+    List<PipelineConfigurable> sinks = new ArrayList<>();
+    for (ETLStage sinkConfig : sinkConfigs) {
+      String sinkPluginId = PluginID.from(Constants.Sink.PLUGINTYPE, sinkConfig.getName(), pluginNum).getID();
+      sinkPluginIds.add(sinkPluginId);
+
+      // try to instantiate the sink
+      PipelineConfigurable sink = configurer.usePlugin(Constants.Sink.PLUGINTYPE, sinkConfig.getName(),
+        sinkPluginId, getPluginProperties(sinkConfig));
+      if (sink == null) {
+        throw new IllegalArgumentException(String.format("No Plugin of type '%s' named '%s' was found.",
```

Comment:
is there anyway users could recover from this error? should we tell them how in the error message?